### PR TITLE
Include registry in the target name

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @bbguimaraes

### CODE REVIEW

File: pkg/steps/release/promote.go [hunk 1]
```diff
@@ -43,11 +43,17 @@ type promotionStep struct {
...
 	if len(config.Name) > 0 {
-		return fmt.Sprintf("%s/%s:${component}", config.Namespace, config.Name)
+		if registry == api.QuayOpenShiftCIRepo {
+			return fmt.Sprintf("%s/%s_%s_${component}", registry, config.Namespace, config.Name)
+		}
+		return fmt.Sprintf("%s/%s/%s:${component}", registry, config.Namespace, config.Name)
+	}
+	if registry == api.QuayOpenShiftCIRepo {
+		return fmt.Sprintf("%s/%s_${component}_%s", registry, config.Namespace, config.Tag)
 	}
```
Comment (bbguimaraes) [inline pkg/steps/release/promote.go:55]: You know what I'm going to say about these `if`s =)  In general, "quay" appearing in this file is a layering violation.

We're also duplicating the logic already present in `pkg/api/promotion.go`.

Comment (hongkailiu) [reply pkg/steps/release/promote.go:55]: Let me try with another way.

File: pkg/api/promotion.go [hunk 2]
```diff
@@ -109,3 +109,17 @@ var (
...
+		if registry == QuayOpenShiftCIRepo {
+			return fmt.Sprintf("%s/%s_%s_${component}", registry, config.Namespace, config.Name)
+		}
+		return fmt.Sprintf("%s/%s/%s:${component}", registry, config.Namespace, config.Name)
+	}
+	if registry == QuayOpenShiftCIRepo {
+		return fmt.Sprintf("%s/%s_${component}_%s", registry, config.Namespace, config.Tag)
+	}
+	return fmt.Sprintf("%s/%s/${component}:%s", registry, config.Namespace, config.Tag)
+}
```
Comment (bbguimaraes) [inline pkg/api/promotion.go:125]: We know in `defaults.go` which registry is used by each step, but loose that information when calling this function.  There is no need for these `if registry == …` conditionals.

File: pkg/defaults/defaults.go [hunk 3]
```diff
@@ -363,18 +363,26 @@ func fromConfig(
...
 		} else {
-			postSteps = append(postSteps, releasesteps.PromotionStep(api.PromotionQuayStepName, config, requiredNames, jobSpec, podClient, pushSecret, api.QuayOpenShiftCIRepo, api.QuayMirrorFunc))
+			postSteps = append(postSteps, releasesteps.PromotionStep(api.PromotionQuayStepName, config, requiredNames, jobSpec, podClient, pushSecret, api.QuayOpenShiftCIRepo, targetName(api.QuayOpenShiftCIRepo, *config.PromotionConfiguration, ":", "_", "_"), api.QuayMirrorFunc))
 		}
 	}
 
 	return append(overridableSteps, buildSteps...), postSteps, nil
 }
 
+func targetName(registry string, config api.PromotionConfiguration, d1, d2, d3 string) string {
```
Comment (hongkailiu) [inline pkg/defaults/defaults.go:379]: d1, d2, d3 are ugly.
But I do not find another way provided you do not want me to do the following check

```go
if registry==... {
}
```

Comment (bbguimaraes) [reply pkg/defaults/defaults.go:379]: Just make it two functions.

**Pull request merged**